src/main/kotlin/Monster.kt

```diff
@@ -5,7 +5,11 @@ class Monster (name: String = "something",
                var health: Int = 5) : Card(name, type, cardId){
 
     fun takeDamge(card: Monster): Boolean {
-        return false
+        if(this.health < 0){return false}
```

Comment:
Can be rewrited to "when" statement 
when(this.health <= 0) {
    true -> false
    false -> {
         this.health = this.health - card.attack
         return this.isDead() // return keyword can be omitted
    }
}